service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/DraftCMOController.java

```diff
@@ -10,33 +10,49 @@
 import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
 import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
 import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
 import uk.gov.hmcts.reform.fpl.model.CaseManagementOrder;
+import uk.gov.hmcts.reform.fpl.service.DirectionHelperService;
 import uk.gov.hmcts.reform.fpl.service.DraftCMOService;
 
-import java.util.Map;
+import static java.util.Objects.isNull;
 
 @Api
 @RestController
 @RequestMapping("/callback/draft-cmo")
 public class DraftCMOController {
     private final ObjectMapper mapper;
     private final DraftCMOService draftCMOService;
+    private final DirectionHelperService directionHelperService;
+
 
     @Autowired
-    public DraftCMOController(ObjectMapper mapper, DraftCMOService draftCMOService) {
+    public DraftCMOController(ObjectMapper mapper,
+                              DraftCMOService draftCMOService,
+                              DirectionHelperService directionHelperService) {
         this.mapper = mapper;
         this.draftCMOService = draftCMOService;
+        this.directionHelperService = directionHelperService;
     }
 
     @PostMapping("/about-to-start")
     public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody CallbackRequest callbackrequest) {
         CaseDetails caseDetails = callbackrequest.getCaseDetails();
-        Map<String, Object> caseData = caseDetails.getData();
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        if (!isNull(caseData.getCaseManagementOrder())) {
```

Comment:
Logic behind that if/else is never tested. How do you know it works?

Reply:
So it's tested in FPLA-657, did not want to repeat something I was going to merge in before story completion.

Reply:
Move this out to a service method to test it properly.